src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobWriter.cs

```diff
@@ -170,7 +170,7 @@ public void WriteBytes(BlobBuilder source)
                 Throw.ArgumentNull(nameof(source));
             }
 
-            source.WriteContentTo(ref this);
+            source!.WriteContentTo(ref this);
```

Comment:
After adding `[DoesNotReturn]` attribute on `Throw.ArgumentNull(string);` `!` not needed